File: Backend/src/main/java/com/tfg/truby_writer/model/services/LocationsService.java
package com.tfg.truby_writer.model.services;

import com.tfg.truby_writer.model.entities.Location;
import com.tfg.truby_writer.model.entities.LocationPoint;
import com.tfg.truby_writer.model.enums.Enums;
import com.tfg.truby_writer.model.services.Block;

import com.tfg.truby_writer.model.exceptions.DuplicateInstanceException;
import com.tfg.truby_writer.model.exceptions.InstanceNotFoundException; 


public interface LocationsService {

    // LOCATIONS

    Location addLocation(Long plotId, String name, String backgroundHistory, String backgroundImageUrl) throws DuplicateInstanceException, InstanceNotFoundException;

    Location modifyLocation(Long locationId, String name, String backgroundHistory, String backgroundImageUrl) throws InstanceNotFoundException;

    void deleteLocation(Long locationId) throws InstanceNotFoundException;

    Location findLocationById(Long locationId) throws InstanceNotFoundException;

    Block<Location> findLocationsByFilter(Long plotId, String name) throws InstanceNotFoundException;

    // LOCATION POINTS
    
    LocationPoint addLocationPoint(Long locationId, String name, String description, Float coordX, Float coordY, Enums.MarkerType markerType, String markerIcon) throws DuplicateInstanceException, InstanceNotFoundException, IllegalArgumentException;
    
    LocationPoint modifyLocationPoint(Long locationPointId, String name, String description, Float coordX, Float coordY, Enums.MarkerType markerType, String markerIcon) throws InstanceNotFoundException, IllegalArgumentException;

    void deleteLocationPoint(Long locationPointId) throws InstanceNotFoundException;

    LocationPoint findLocationPointById(Long locationPointId) throws InstanceNotFoundException;


    }

File: Frontend/src/backend/locationsService.js
import { appFetch } from '../../backend/appFetch';

const BASE_PATH = '/structure';


export const addLocation = async (plotId, locationDto) => {
    return await appFetch('POST', `${BASE_PATH}/plots/${plotId}/createLocation`, locationDto);
};


export const modifyLocation = async (locationId, locationDto) => {
    return await appFetch('PUT', `${BASE_PATH}/modify/${locationId}`, locationDto);
};


export const deleteLocation = async (locationId) => {
    return await appFetch('DELETE', `${BASE_PATH}/delete/${locationId}`);
};


export const findLocationById = async (locationId) => {
    return await appFetch('GET', `${BASE_PATH}/${locationId}`);
};


export const findLocationsByFilter = async (plotId, name) => {
    return await appFetch('GET', `${BASE_PATH}/plots/${plotId}/search?name=${encodeURIComponent(name)}`);
};


export const addLocationPoint = async (locationId, locationPointDto) => {
    return await appFetch('POST', `${BASE_PATH}/${locationId}/createPoint`, locationPointDto);
};


export const modifyLocationPoint = async (locationPointId, locationPointDto) => {
    return await appFetch('PUT', `${BASE_PATH}/points/modify/${locationPointId}`, locationPointDto);
};


export const deleteLocationPoint = async (locationPointId) => {
    return await appFetch('DELETE', `${BASE_PATH}/points/delete/${locationPointId}`);
};


export const findLocationPointById = async (locationPointId) => {
    return await appFetch('GET', `${BASE_PATH}/points/${locationPointId}`);
};